mypy/newsemanal/semanal.py

```diff
@@ -1899,48 +1946,113 @@ def is_not_ready_type_ref(self, rv: Expression) -> bool:
                     return True
         return False
 
+    def should_wait(self, rv: Expression) -> bool:
+        """Can we already classify this r.h.s. of an assignment or should wait?
+
+        This returns True if we don't have enough information to decide whether
+        an assignment is just a normal variable definition or a special form.
+        Always return False if this is a final iteration, this will typically cause
+        the assignment to be classified as variable plus emmit an error.
```

Comment:
Typo: emmit -> emit

Maybe change 'the assignment' -> 'the lvalue'.

Grammar nit: 'variable' -> 'a variable'